server/app/repository.py

```diff
@@ -71,6 +85,12 @@ def exclude_from_dict(d, keys: Sequence[str]):
 
     return [obj,] + excluded
 
+class DefaultDictWithAnyKey(defaultdict):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+    def __contains__(self, item):
+        return True
```

Comment:
Huh?

Reply:
First - theory. We use defaultdict as base class. If you try to access key which not exists then you get default value. But if you check existence of non-existing key then you get False.

Second - practice. Idea of use defaultdict was to handle not-provided query parameters without list them. If user doesn't provide value of any parameters then we want to use None value. We pass query and parameters dict to external (psycopg2) function.
When external function process dict then it check of existence of each needed key. If we will to use standard defaultdict then it returns False on check, but it is able to return value.

This class change this behavior and returns True if you check existence of any key. Therefore it is named "DefaultDictWithAnyKey".

Reply:
I added docstring to this class.